src/DynamoDbQueryBuilder.php

```diff
@@ -154,13 +155,34 @@ public function find($id, array $columns = [])
         return $model;
     }
 
+    public function findOrFail($id, array $columns = [])
+    {
+        $result = $this->find($id, $columns);
+        
+        if (is_array($id) && count($result) == count(array_unique($id))) {
+            return $result;
+        } elseif (! is_null($result)) {
```

Comment:
Oh, I meant to say let's just copy over this chunk https://github.com/laravel/framework/blob/1742bc9a378162c3b57f1bc799e91f2ab0802aef/src/Illuminate/Database/Eloquent/Builder.php#L300-L315 as is. Your version is a slightly modified version of it.